API-Reference/Kokkos::Subview.md

```diff
@@ -0,0 +1,42 @@
+# `Kokkos::Subview`
+
+Header File: `Kokkos_Core.hpp`
+
+Alias template to deduce the type that is returned by a call to the subview function with given arguments.
+
+Usage:
+  ```c++
+  Kokkos::Subview<ViewType,Args> subView;
+  ```
+
+## Description
+
+  ```c++
+  template <class ViewType, class... Args>
+  using Subview = IMPL_DETAIL; // deduce subview type from source view traits
+  ```
+  Type of a `Kokkos::View` viewing a subset of `ViewType` specified by `Args...`.
+  Same type as returned by a call to the subview function with corresponding arguments.
+  For restrictions on Args see subview documentation.
+   
+## Examples
+
+```c++
+
+  using view_type = Kokkos::View<double ***[5]>;
+  view_type a("A",N0,N1,N2);
+
+  struct subViewHolder {
+  Kokkos::Subview<view_type,
+                  std::pair<int,int>,
+                  int,
+                  Kokkos::ALL,int> s;
+  } subViewHolder;
+
+  subViewHolder.s  = Kokkos::subview(a,
+                                     std::pair<int,int>(3,15),
+                                     5,
+                                     Kokkos::ALL,
+                                     Kokkos::ALL);
+
+```
```

Comment:
The documentation you guys wrote is good.  I am just not convinced we should provide `Subview` in the first place considering we have `decltype`.

Reply:
yeah but decltype + declval is somewhat a pain to use.

Reply:
How is
```C++
using subview_type = decltype(Kokkos::subview(std::declval<view_type>(), std::make_pair(1, 2), 3, Kokkos::ALL));
```
worse than
```C++
using subview_type = Kokkos::Subview<view_type, Kokkos::pair<int, int>, std::pair<int, int>, int, Kokkos::ALL>;
```
?
You use native C++ construct rather than having to lookup yet another Kokkos class template.

Reply:
Hmm I think as long as it is in our public API we should have at least minimal documentation on it or deprecate it.
As the discussion goes: I see no disadvantages of keeping it in the public API even though it can be achieved by decltype and declval ... we basically just directly expose the type that subview is deducing